pir/cpp/client.cpp

```diff
@@ -35,19 +35,37 @@ using ::seal::Plaintext;
 using ::seal::RelinKeys;
 
 PIRClient::PIRClient(std::unique_ptr<PIRContext> context)
-    : context_(std::move(context)) {
-  auto sealctx = context_->SEALContext();
-  keygen_ = std::make_unique<seal::KeyGenerator>(sealctx);
-  encryptor_ =
-      std::make_shared<seal::Encryptor>(sealctx, keygen_->public_key());
-  decryptor_ =
-      std::make_shared<seal::Decryptor>(sealctx, keygen_->secret_key());
+    : context_(std::move(context)) {}
+
+Status PIRClient::initialize() {
+  ASSIGN_OR_RETURN(db_, PIRDatabase::Create(context_->Params()));
+  try {
+    auto sealctx = context_->SEALContext();
+    keygen_ = std::make_unique<seal::KeyGenerator>(sealctx);
+    encryptor_ =
+        std::make_shared<seal::Encryptor>(sealctx, keygen_->public_key());
+    decryptor_ =
+        std::make_shared<seal::Decryptor>(sealctx, keygen_->secret_key());
+    auto gal_keys = keygen_->galois_keys(generate_galois_elts(
+        context_->EncryptionParams().poly_modulus_degree()));
+    auto relin_keys = keygen_->relin_keys_local();
```

Comment:
This function doesn't return the serializable type. Not sure if it affects performance that much, but it might be worth it to keep the interface the same now that we have changed the type of Galois Keys

Reply:
Good catch, I'd meant to change this one too. Thanks!